### PR TITLE
podcasts

### PR DESCRIPTION
**Pull request by i-like-robots**
- Fetches multiple RSS feeds for each show in series
- For each feed it will take each episode, map it to the (V1) article model and insert them into ES in throttled batches
- It maps each episode to meta data and old Podhoster IDs to new Acast GUIDs
- It will attempt to update each episode up to 3 times
- It can reindex all episodes or only add new shows

screen shot 2015-09-03 at 12 50 05

Depends on https://github.com/Financial-Times/next-feature-flags-api/pull/326 and https://github.com/Financial-Times/next-ft-api-client/pull/48

### CODE REVIEW

File: server/podcast-feeds.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+var args = require('minimist')(process.argv.slice(2));
+var PodcastUpdater = require('./processes/podcast-updater');
+
+var feeds = [
+	'http://rss.acast.com/ft-uk-general-election-2015'
+];
+
+var tasks = feeds.map(function(feed) {
+	return new PodcastUpdater({
+		elasticSearchURL: 'https://b1befb2bddf0fd14d4a5285ccd141646.eu-west-1.aws.found.io:9243/',
```
Comment (i-like-robots) [inline server/podcast-feeds.js]: This is my own cluster for testing

**Comment by matthew-andrews** [comment]
One small question, have you checked what happens when the continuous updater encounters a podcast?

**Comment by i-like-robots** [comment]
No, added to to-do

File: server/processes/podcast-updater.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			if (data.hits.total) {
+				return new Date(data.hits.hits[0]._source._lastUpdatedDateTime);
+			}
+		});
+};
+
+PodcastUpdater.prototype.fetchRssFeed = function() {
+	var parser = new FeedMe(true);
+
+	return fetch(this.options.rssFeedUrl, { timeout: 3000 })
```
Comment (matthew-andrews) [inline server/processes/podcast-updater.js]: error handling?

File: server/processes/podcast-updater.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+		.then(function(response) {
+			return response.text();
+		})
+		.then(function(text) {
+			parser.on('error', function(err) {
+				throw err;
```
Comment (matthew-andrews) [inline server/processes/podcast-updater.js]: oh i guess that'll do

File: server/processes/podcast-updater.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+				throw new Error(data.error);
+			}
+
+			console.log('' + id + ' updated to version ' + data._version + '.');
+			return Promise.resolve();
+		});
+};
+
+PodcastUpdater.prototype.error = function(error) {
+	console.error(error.message || error);
```
Comment (matthew-andrews) [inline server/processes/podcast-updater.js]: ^^ `raven.captureError` ?

**Comment by matthew-andrews** [comment]
Looks good…  it would be good to think about how we monitor these to make sure they're working.  E.g. the health check for CAPI content will fail if it can't continuously update… a good test of quite a lot of the moving parts as we learned this week……

[Not necessary for this stage, but probably ought to be done before the migration from FT to Next is done]

File: server/processes/podcast-updater.js (hunk 4 above)
Comment (i-like-robots) [inline server/processes/podcast-updater.js]: Pardon?

Comment (matthew-andrews) [inline server/processes/podcast-updater.js]: can we send these these errors to sentry :innocent:

Comment (matthew-andrews) [inline server/processes/podcast-updater.js]: or rather, should we?

**Comment by i-like-robots** [comment]
Current plan is to run it manually a few times to verify the data is OK. Health checks and setting up the scheduler could be next?

File: server/processes/podcast-updater.js (hunk 4 above)
Comment (i-like-robots) [inline server/processes/podcast-updater.js]: I think we should, maybe as mentioned below I can add monitoring etc. when this task requires scheduling?

**Comment by matthew-andrews** [comment]
sounds good

**Pull request merged**